contracts/babylon/src/contract.rs

```diff
@@ -269,19 +269,21 @@ pub fn execute(
     match msg {
         ExecuteMsg::Slashing { evidence } => {
             // This is an internal routing message from the `btc_finality` contract
+            // following https://github.com/babylonlabs-io/babylon/blob/4aa85a8d9bf85771d448cd3026e99962fe0dab8e/x/finality/keeper/msg_server.go#L384-L413 without the logic for propagating the slashing event to other BSNs
```

Comment:
Could you also remind me why we send the slashing message to the babylon contract and then the staking contract instead of sending it to the staking contract directly?

Reply:
This is because the finality contract relies on Babylon contract for sending IBC packets about slashing to Babylon. The finality contract does not have access to the zone concierge IBC channel.

Reply:
There are two purposes here: one is to send the slashing message for the staking contract for processing, the other is to send the slashing message to Babylon Genesis. Currently, the finality sends one slashing message to the finality contract, and then the finality contract sends a message to the staking contract as well as Babylon Genesis. I mean why not send out two messages from the finality contract, one sent to the staking contract directly, one sent to the babylon contract to propagate it to Babylon Genesis. That being said, the path of sending the slashing message to the staking contract can be done more efficiently, right?

Reply:
> I mean why not send out two messages from the finality contract, one sent to the staking contract directly, one sent to the babylon contract to propagate it to Babylon Genesis. That being said, the path of sending the slashing message to the staking contract can be done more efficiently, right?

Yes this is a potential improvement indeed